horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/user/DatabaseUserBackend.java

```diff
@@ -127,34 +139,44 @@ private void addTeamMembership(UserInfo userInfo, String teamName, TeamRole role
         });
     }
 
+    @Transactional
     @WithRoles(extras = Roles.HORREUM_SYSTEM)
     @Override public List<String> getAllTeams() {
         List<Team> teams = Team.listAll();
-        return teams.stream().map(t -> t.teamName + "-team").collect(toList());
+        return teams.stream().map(t -> t.teamName + "-team").toList();
     }
 
     @Transactional
-    @WithRoles(extras = Roles.HORREUM_SYSTEM)
     @Override public void addTeam(String team) {
-        String prefix = getTeamPrefix(team); // perform validation of the team name
-        Team.getEntityManager().merge(new Team(prefix.substring(0, prefix.length() - 1)));
+        Team.getEntityManager().merge(new Team(removeTeamSuffix(team)));
     }
 
     @Transactional
     @WithRoles(extras = Roles.HORREUM_SYSTEM)
     @Override public void deleteTeam(String team) {
-        String prefix = getTeamPrefix(team);
-        Team teamEntity = Team.find("teamName", prefix.substring(0, prefix.length() - 1)).firstResult();
+        Team teamEntity = Team.find("teamName", removeTeamSuffix(team)).firstResult();
         if (teamEntity == null) {
-            throw ServiceException.notFound("The team " + team + " does not exist");
+            throw ServiceException.notFound(format("The team {0} does not exist", team));
         }
-        TeamMembership.delete("team", teamEntity);
+        // need to delete memberships with roles
+        deleteTeamAndMemberships(teamEntity);
```

Comment:
It this throws an exception it would be propagated to the client directly if I'm not mistaken. If so we should add some logic to capture that.